screen/categories/src/main/java/com/ivy/categories/CategoriesViewModel.kt

```diff
@@ -171,9 +183,21 @@ class CategoriesViewModel @Inject constructor(
             }
 
             val sortedList = sortList(categories, sortOrder.value).toImmutableList()
+            this.allCategories.value = sortedList
+            this.categories.value = this.allCategories.value
+        }
+    }
 
-            this.categories.value = sortedList
+    private fun filterCategories(queryString: String) {
+        var unsortedList: List<CategoryData>
+        if (queryString.isNotEmpty()) {
+            unsortedList = this.allCategories.value.filter {
+                it.category.name.value.toLowerCase().contains(queryString.toLowerCase().trim())
```

Comment:
It should be:
```kotlin
query.lowercase().trim() in it.category.name.lowercase()
```

Logically, we're looking of the existence of a keyword/query in a category name and not the other way around

Reply:
Done